app/common/model/src/main/java/io/syndesis/common/model/DataShape.java

```diff
@@ -124,7 +121,7 @@ public Builder decompress() {
                 } catch (IOException e) {
                     throw new IllegalStateException("Failed to decompress data shape", e);
                 }
-            } else if (currentShape.getVariants().stream().anyMatch(variant -> "true".equals(variant.getMetadata().get(COMPRESSION_METADATA_KEY)))) {
+            } else if (currentShape.getVariants().stream().anyMatch(variant -> "true".equals(variant.getMetadata().get(DataShapeMetaData.COMPRESSION)))) {
```

Comment:
```suggestion
            } else if (currentShape.getVariants().stream().anyMatch(variant -> Boolean.parseBoolean(variant.getMetadata().get(DataShapeMetaData.COMPRESSION)))) {
```

Though I would probably pull this in a helper method like `isCompressed(DataShape dataShape)`

Reply:
Yeah that `isCompressed()` helper is a good idea